Amazon.QLDB.Driver.IntegrationTests/AsyncStatementExecutionTests.cs

```diff
@@ -0,0 +1,144 @@
+    using System.Text;
+    using System.Threading.Tasks;
+
+    [TestClass]
+    public class StatementExecutionTests
+    {
+        private static readonly IValueFactory ValueFactory = new ValueFactory();
+        private static AmazonQLDBSessionConfig amazonQldbSessionConfig;
+        private static AsyncIntegrationTestBase integrationTestBase;
+        private static AsyncQldbDriver qldbDriver;
+
+        [ClassInitialize]
+        public async static void SetUp(TestContext context)
+        {
+            // Get AWS configuration properties from .runsettings file.
+            string region = context.Properties["region"].ToString();
+
+            amazonQldbSessionConfig = AsyncIntegrationTestBase.CreateAmazonQLDBSessionConfig(region);
+            integrationTestBase = new AsyncIntegrationTestBase(Constants.LedgerName, region);
+
+            integrationTestBase.RunForceDeleteLedger();
+
+            integrationTestBase.RunCreateLedger();
+            qldbDriver = integrationTestBase.CreateDriver(amazonQldbSessionConfig);
+
+            // Create table.
+            var query = $"CREATE TABLE {Constants.TableName}";
+
+            var count = await qldbDriver.Execute(async txn =>
+            {
+                var result = await txn.Execute(query);
+
+                var count = 0;
+                await foreach (var row in result)
+                {
+                    count++;
+                }
+                return count;
+            });
+
+
+            Assert.AreEqual(1, count);
+
+            var result = await qldbDriver.ListTableNamesAsync();
+            foreach (var row in result)
+            {
+                Assert.AreEqual(Constants.TableName, row);
+            }
+        }
+
+        [ClassCleanup]
+        public static void ClassCleanup()
+        {
+            integrationTestBase.RunDeleteLedger();
+            qldbDriver.Dispose();
+        }
+
+        [TestCleanup]
+        public async void TestCleanup()
+        {
+            // Delete all documents in table.
+            await qldbDriver.Execute(txn => txn.Execute($"DELETE FROM {Constants.TableName}"));
+        }
+
+        [TestMethod]
+        public async void Execute_DropExistingTable_TableDropped()
+        {
+            // Given.
+            var create_table_query = $"CREATE TABLE {Constants.CreateTableName}";
+            var create_table_count = await qldbDriver.Execute(async txn =>
+            {
+                var result = await txn.Execute(create_table_query);
+
+                var count = 0;
+                await foreach (var row in result)
+                {
+                    count++;
+                }
+                return count;
+            });
+            Assert.AreEqual(1, create_table_count);
+
+            // Execute ListTableNames() to ensure table is created.
+            var result = await qldbDriver.ListTableNamesAsync();
+
+            var tables = new List<string>();
+            foreach (var row in result)
```

Comment:
await

Reply:
Since 'result' object here returned by 'await qldbDriver.ListTableNamesAsync();' is an IEnumerable<string>, we cannot do: await foreach (var row in result).  This is unless qldbDriver.ListTableNamesAsync() changes its return value in future PRs.